keel-artifact/src/main/kotlin/com/netflix/spinnaker/keel/artifact/ArtifactListener.kt

```diff
@@ -57,23 +49,15 @@ class ArtifactListener(
       .artifacts
       .filter { it.type.toLowerCase() in artifactTypeNames }
       .forEach { artifact ->
-        if (repository.isRegistered(artifact.name, artifact.type())) {
-          val version: String
-          var status: ArtifactStatus? = null
-          when (artifact.type()) {
-            deb -> {
-              version = "${artifact.name}-${artifact.version}"
-              status = debStatus(artifact)
-            }
-            docker -> {
-              version = artifact.version
-            }
-          }
+        if (repository.isRegistered(artifact.name, artifact.artifactType)) {
```

Comment:
`ArtifactListener` now has no artifact type-specific logic -- every piece of information needed is retrieved from the `ArtifactPublisher` matching the artifact type via the newly defined functions on the interface.